frontend/public/components/utils/get-pod-traffic.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Tooltip } from '@patternfly/react-core';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { EndpointSliceKind } from '@console/internal/module/k8s';
+import { ConnectedIcon, DisconnectedIcon } from '@patternfly/react-icons';
+import { EndPointSliceModel } from '@console/app/src/models';
+
+export type GetPodTrafficProp = {
+  podName: string;
+  namespace: string;
+  tooltipFlag?: boolean;
+};
+
+export const GetPodTraffic: React.FC<GetPodTrafficProp> = ({ podName, namespace, tooltipFlag }) => {
+  const { t } = useTranslation();
+  const [data, loaded, loadError] = useK8sWatchResource<EndpointSliceKind[]>({
+    groupVersionKind: {
+      kind: EndPointSliceModel.kind,
+      version: EndPointSliceModel.apiVersion,
+    },
+    isList: true,
+    namespaced: true,
+    namespace,
+  });
+
+  const allEndpoints = data.reduce((prev, next) => prev.concat(next?.endpoints), []);
```

Comment:
wold be better to check if loaded, no load error, and if data is present before processing?